lib/model/folder.go

```diff
@@ -327,11 +324,7 @@ func (f *folder) restartWatch() {
 func (f *folder) startWatch() {
 	ctx, cancel := context.WithCancel(f.ctx)
 	f.model.fmut.RLock()
-	ignores, ok := f.model.folderIgnores[f.folderID]
-	if !ok {
-		l.Debugf("Failed to get ignore patterns for folder %v while starting the filesystem watcher -> aborting, folder will be stopped soon (racy)", f.Description())
-		return
-	}
+	ignores := f.model.folderIgnores[f.folderID]
```

Comment:
We should however add a small delay to the timer, so that if both the context is cancelled, and the timer is fired, the ctx.Done() would win.

Reply:
I don't think it's necessary: Due to `RemoveAndWait` `folderIgnores` won't be removed until this is finished, so it can't be `nil`.

Reply:
Right.